test/jasmine/tests/command_test.js

```diff
@@ -0,0 +1,641 @@
+            }, [1]]);
+
+            expect(result).toEqual([
+                {type: 'data', prop: 'y', traces: [1], value: [[3, 4, 5]]},
+                {type: 'data', prop: 'marker.size', traces: [1], value: [10]},
+                {type: 'data', prop: 'line.color', traces: [1], value: ['red']},
+                {type: 'data', prop: 'line.width', traces: [1], value: [2]},
+                {type: 'layout', prop: 'margin.l', value: 50},
+                {type: 'layout', prop: 'width', value: 10}
+            ]);
+        });
+    });
+
+    describe('animate', function() {
+        it('binds to the frame for a simple animate command', function() {
+            var result = Plots.computeAPICommandBindings(gd, 'animate', [['framename']]);
+
+            expect(result).toEqual([{type: 'layout', prop: '_currentFrame', value: 'framename'}]);
+        });
+
+        it('binds to nothing for a multi-frame animate command', function() {
+            var result = Plots.computeAPICommandBindings(gd, 'animate', [['frame1', 'frame2']]);
+
+            expect(result).toEqual([]);
+        });
+    });
+});
+
+describe('component bindings', function() {
+    'use strict';
+
+    var gd;
+    var mock = require('@mocks/binding.json');
+
+    beforeEach(function(done) {
+        var mockCopy = Lib.extendDeep({}, mock);
+        gd = createGraphDiv();
+
+        Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(done);
+    });
+
+    afterEach(function() {
+        destroyGraphDiv(gd);
+    });
+
+    it('creates an observer', function(done) {
+        var count = 0;
+        Plots.manageCommandObserver(gd, {}, [
+            { method: 'restyle', args: ['marker.color', 'red'] },
+            { method: 'restyle', args: ['marker.color', 'green'] }
+        ], function(data) {
+            count++;
+            expect(data.index).toEqual(1);
+        });
+
+        // Doesn't trigger the callback:
+        Plotly.relayout(gd, 'width', 400).then(function() {
+            // Triggers the callback:
+            return Plotly.restyle(gd, 'marker.color', 'green');
+        }).then(function() {
+            // Doesn't trigger a callback:
+            return Plotly.restyle(gd, 'marker.width', 8);
+        }).then(function() {
+            expect(count).toEqual(1);
+        }).catch(fail).then(done);
+    });
+
+    it('logs a warning if unable to create an observer', function() {
+        var warnings = 0;
+        spyOn(Lib, 'warn').and.callFake(function() {
+            warnings++;
+        });
+
+        Plots.manageCommandObserver(gd, {}, [
+            { method: 'restyle', args: ['marker.color', 'red'] },
+            { method: 'restyle', args: [{'line.color': 'green', 'marker.color': 'green'}] }
+        ]);
+
+        expect(warnings).toEqual(1);
+    });
+
+    it('udpates bound components when the value changes', function(done) {
+        expect(gd.layout.sliders[0].active).toBe(0);
+
+        Plotly.restyle(gd, 'marker.color', 'blue').then(function() {
```

Comment:
Ah, confused it with the deluxe tests down below. Yeah, these are just nice and simple 😄